Lib/test/test_unpack.py

```diff
@@ -167,5 +219,47 @@ def test_extended_oparg_not_ignored(self):
             y = unpack_400(range(400))
             self.assertEqual(y, 399)
 
+    def test_exception_context_when_len_fails(self):
+        """When `__len__()` raises an Exception, preserve exception context"""
+        code = textwrap.dedent(
+            f"""
+            class CustomSeq:
+                def __len__(self):
+                    raise RuntimeError
+
+                def __getitem__(self, i):
+                    return i*2
+
+            x, y, z = CustomSeq()
+            """
+        )
+        try:
+            exec(code)
+        except ValueError as exc:
+            self.assertEqual(exc.args, ('too many values to unpack (expected 3)',))
+            self.assertIsInstance(exc.__context__, RuntimeError)
```

Comment:
There's [a helper for that](https://docs.python.org/3.14/library/unittest.html#unittest.TestCase.assertRaises):
```suggestion
        with self.assertRaises(RuntimeError) as cm:
            exec(code)

        self.assertEqual(str(cm.exception), 'too many values to unpack (expected 3)')
        self.assertIsInstance(cm.exception.__context__, RuntimeError)
```
(untested code)

Reply:
My bad! I thought `pytest` invented that one 😄